### ISSUE TITLE
avg_reward becomes negative when using new self critical

### ISSUE THREAD

**Issue by xuewyang** [ruotianluo/self-critical.pytorch#188]
Hi Ruotian,
Thank you for your well implemented codes. Can you help with the following questions?
1. When I am using updown_nsc.yml for training, the avg_reward outputs negative values. Is this normal?
2. data/cocotalk_box is not used for training right?
Thank you.

**Comment by ruotianluo** [comment]
1. It's normal
2. Yes.

**Comment by xuewyang** [comment]
TY

**Comment by ruotianluo** [comment]
Wait. If you are using new self critical, there should not be avg_reward. You can checkout the code. That value appears only using self critical.

**Comment by xuewyang** [comment]
I see. In updown_nsc.yml, I used your default settings:
self_critical_after: -1
structure_after: 30
Probaly I have to set self_critical_after = 30. I have a question. How to set up self_critical_after and structure_after, these two parameters? When do we use structure training?

**Comment by ruotianluo** [comment]
The yaml file is correct. New-self-critical is a kind of structure loss (at least in the code I treat it as one).

**Comment by xuewyang** [comment]
Oh, my bad. I used the wrong yml. Now I am using the correct one and having the following output.
iter 14560 (epoch 30), train_loss = -0.008, lm_loss = 0.000, struc_loss = -0.008, time/batch = 3.451

When doing SC or NSC, you are not training with Xentropy, right?

**Comment by ruotianluo** [comment]
No. That's why lm_loss is always 0.
